### ISSUE TITLE
partial connect non-share-weight layer

### ISSUE THREAD

**Issue by xyzhang16** [fchollet/keras#3050]
Hi
I want to define a partial connect layer simalar to convolutional network, but it is non-weight sharing.
Is there any good idea?

**Comment by xyzhang16** [comment]
code  is following:
`
import numpy as np
from keras import backend as K
from keras.engine import  Layer

class PartialDense(Layer):

    def __init__(self, input_dim,size=2,  **kwargs):
        self.size=size
        self.output_dim = input_dim-size+1  # fix
        self.input_dim = input_dim
        kwargs['input_shape'] = (self.input_dim,)
        super(PartialDense, self).__init__(**kwargs)

    def build(self, input_shape):
        initial_weight_value=np.random.random((self.size,self.output_dim))
        self.W = K.variable(initial_weight_value)
        self.trainable_weights = [self.W]

    def call(self, x, mask=None):
        list=[]
        for i in range(self.output_dim):
            valve=K.dot(x[i:i+self.size],K.transpose(self.W)[i])
            list.append(valve)
        output=K.concatenate(list)
        return output

    def get_output_shape_for(self, input_shape):
        return (self.input_dim,self.output_dim)
`
why is it wrong?
Exception: Input 0 is incompatible with layer dense_1: expected ndim=2, found ndim=1
